### PR TITLE
[SPARK-18143][SQL]Ignore Structured Streaming event logs to avoid breaking history server (branch 2.0)

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Backport #15663 to branch-2.0 and fixed conflicts in `ReplayListenerBus`.

## How was this patch tested?

Jenkins

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas @rxin @marmbrus @yhuai

**Comment by rxin** [comment]
LGTM pending Jenkins.

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryListenerSuite.scala [hunk 1]
```diff
@@ -206,6 +209,45 @@ class StreamingQueryListenerSuite extends StreamTest with BeforeAndAfter {
     assert(queryQueryTerminated.exception === newQueryTerminated.exception)
   }
 
+  test("ReplayListenerBus should ignore broken event jsons generated in 2.0.0") {
+    // query-event-logs-version-2.0.0.txt has all types of events generated by
+    // Structured Streaming in Spark 2.0.0.
+    // SparkListenerApplicationEnd is the only valid event and it's the last event. We use it
+    // to verify that we can skip broken jsons generated by Structured Streaming.
+    testReplayListenerBusWithBorkenEventJsons("query-event-logs-version-2.0.0.txt")
```
Comment (JoshRosen) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryListenerSuite.scala]: Typo: `Borken` -> `Broken`

Comment (zsxwing) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryListenerSuite.scala]: > Typo: Borken -> Broken

Fixed

**Comment by zsxwing** [comment]
Looks like FileStreamSourceSuite is broken is 2.0. Looking at it.

**Comment by zsxwing** [comment]
Fixing the flaky test in #15699.

Retest this please

**Comment by rxin** [comment]
Merging in branch-2.0. Can you close this?

**Comment by zsxwing** [comment]
Sure.

**Pull request closed without merging**